Aplib.Core/Intent/Actions/QueryAction.cs

```diff
@@ -29,20 +29,26 @@ public class QueryAction<TBeliefSet, TQuery> : Action<TBeliefSet>, IQueryable<TB
         /// <summary>
         /// Initializes a new instance of the <see cref="QueryAction{TBeliefSet,TQuery}" /> class.
         /// </summary>
-        /// <param name="effect">The effect of the action.</param>
-        /// <param name="query">The query of the action.</param>
         /// <param name="metadata">
         /// Metadata about this action, used to quickly display the action in several contexts.
         /// </param>
-        public QueryAction(System.Action<TBeliefSet, TQuery> effect,
-            System.Func<TBeliefSet, TQuery?> query,
-            Metadata? metadata = null)
+        /// <param name="effect">The effect of the action.</param>
+        /// <param name="query">The query of the action.</param>
+        public QueryAction
+            (Metadata metadata, System.Action<TBeliefSet, TQuery> effect, System.Func<TBeliefSet, TQuery?> query)
             : base(metadata)
```

Comment:
Ties into previous comment.
```suggestion
            : base(metadata, _ => {})
```